### PR TITLE
Add isogram exercise

### PR DESCRIPTION
**Pull request by ricemery**
Add isogram exercise

### CODE REVIEW

File: exercises/isogram/example.scala [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import scala.collection.mutable
+
+object Isogram {
+  def isIsogram(s: String): Boolean = {
+    val seen = mutable.HashSet[Char]()
```
Comment (abo64) [inline exercises/isogram/example.scala]: Shall we prefer an FP solution?  
I always advice against using mutable objects and `return` if there is an alternative FP solution, so I am forced by my inner Scala-self to do it here, too. :)

And now that I have a clean conscience: I know that the example solution is not essential. You may just forget about this comment after all. :)

**Comment by ricemery** [comment]
Thanks @abo64 .

I have submitted, what I think, is a cleaner example.

File: exercises/isogram/example.scala [hunk 2]
```diff
@@ -0,0 +1,8 @@
+object Isogram {
+  def isIsogram(s: String): Boolean = {
+    !s.filter(_.isLetter)
+      .map(_.toLower)
+      .groupBy(c => c)
```
Comment (abo64) [inline exercises/isogram/example.scala]: Use `identity`?

File: exercises/isogram/example.scala [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+      .values.exists(xs => xs.length > 1)
```
Comment (abo64) [inline exercises/isogram/example.scala]: Apply eta-reduction?

**Comment by ricemery** [comment]
Thanks @abo64 ! I have implemented your suggestions.

**Comment by abo64** [comment]
thanks @ricemery

**Pull request merged**